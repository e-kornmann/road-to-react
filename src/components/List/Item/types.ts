import { Article } from '../../../types';

export type ArticleProps = {
  item: Article;
  onRemoveItem: (item: Article) => void;
};
