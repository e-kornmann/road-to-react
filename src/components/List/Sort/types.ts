export type SortProps = {
  activeSort: string;
  clickHandler: (sortKey: string) => void;
  isReversedOrder: boolean;
  isMediumDevice: boolean;
};
