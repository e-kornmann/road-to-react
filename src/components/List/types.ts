import { Article } from '../../types';

export type ListProps = {
  list: Article[];
  onRemoveItem: (item: Article) => void;
  isMediumDevice: boolean;
  isLargeDevice: boolean;
};
