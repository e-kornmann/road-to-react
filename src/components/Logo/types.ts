export type Props = {
  isLargeDevice: boolean;
};
