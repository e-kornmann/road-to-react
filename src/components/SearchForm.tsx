import React from 'react'
import InputWithLabel from './InputWithLabel';
import StyledSearchForm from './styledComponents/StyledSearchForm';
import * as S from './styledComponents/StyledButtons';

type SearchFormProps = {
    searchTerm: string,
    handleSearch: (event: React.ChangeEvent<HTMLInputElement>) => void,
    handleSearchSubmit:  (event: React.FormEvent<HTMLFormElement>) => void,
    isMediumDevice: boolean,
 
}


const SearchForm = ({ searchTerm, handleSearch, handleSearchSubmit, isMediumDevice }: SearchFormProps) => {
  

  return (
    <StyledSearchForm onSubmit={handleSearchSubmit}>
      <InputWithLabel
        id="search"
        value={searchTerm}
        isFocused
        onInputChange={handleSearch}
      >
      <span>Search:</span>
      </InputWithLabel>
      { isMediumDevice 
        ? (<S.StyledButtonSmall
            type="submit"
            disabled={!searchTerm}  
           >
          Submit
          </S.StyledButtonSmall>) : (
            <S.StyledButtonLarge
            type="submit"
            disabled={!searchTerm}  
           >
          Submit
          </S.StyledButtonLarge>)
          
      }
      
    </StyledSearchForm>
  )
}

export default SearchForm
