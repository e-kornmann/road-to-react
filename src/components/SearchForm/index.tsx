import InputWithLabel from './InputWithLabel/InputWithLabel';
import StyledSearchForm from './style';
import * as S from '../shared/StyledButtons';
import { SearchFormProps } from './types';

const SearchForm = ({
  searchTerm,
  onSearchInput,
  onSearchSubmit,
  isMediumDevice,
  hide,
}: SearchFormProps) => (
    <StyledSearchForm onSubmit={onSearchSubmit}>
      <InputWithLabel
        id="search"
        value={searchTerm}
        isFocused
        onInputChange={onSearchInput}
        hide={hide}
      >
        <span>Search:</span>
      </InputWithLabel>
      {isMediumDevice ? (
        <S.StyledButtonSmall data-testid="submit-button-small" type="submit" disabled={!searchTerm}>
          Submit
        </S.StyledButtonSmall>
      ) : (
        <S.StyledButtonLarge data-testid="submit-button-large" type="submit" disabled={!searchTerm}>
          Submit
        </S.StyledButtonLarge>
      )}
    </StyledSearchForm>
);

export default SearchForm;
