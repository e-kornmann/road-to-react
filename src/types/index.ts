type Article = {
  title: string;
  url: string;
  author: string;
  num_comments: number;
  points: number;
  objectID: number;
  created_at: string;
};

  type StoriesState = {
    data: Article[];
    isLoading: boolean;
    isError: boolean;
  };

  type StoriesFetchInitAction = {
    type: 'STORIES_FETCH_INIT';
  };

  type StoriesFetchSuccessAction = {
    type: 'STORIES_FETCH_SUCCESS';
    payload: Article[];
  };

  type StoriesFetchFailureAction = {
    type: 'STORIES_FETCH_FAILURE';
  };

  type StoriesRemoveAction = {
    type: 'REMOVE_STORY';
    payload: Article;
  };

  type StoriesAction =
    | StoriesFetchInitAction
    | StoriesFetchSuccessAction
    | StoriesFetchFailureAction
    | StoriesRemoveAction;

export type {
  Article,
  StoriesState,
  StoriesFetchSuccessAction,
  StoriesFetchFailureAction,
  StoriesRemoveAction,
  StoriesAction,
};
