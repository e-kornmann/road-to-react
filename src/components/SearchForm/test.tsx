import {
  describe,
  it,
  expect,
  vi,
} from 'vitest';
import {
  render,
  screen,
  fireEvent,
} from '@testing-library/react';
import SearchForm from '../SearchForm';
import { SearchFormProps } from './types';

describe('SearchForm', () => {
  const searchProps: SearchFormProps = {
    searchTerm: 'React',
    onSearchInput: vi.fn(),
    onSearchSubmit: vi.fn(),
    isMediumDevice: true,
  };

  it('renders the input field with React as its initial value', () => {
    render(<SearchForm {...searchProps} />);
    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveValue('React');
  });

  it('renders the small submit button on medium devices', () => {
    render(<SearchForm {...searchProps} />);
    const submitButton = screen.getByRole('button', { name: 'Submit' });
    expect(submitButton).toBeInTheDocument();
    expect(screen.getByTestId('submit-button-small')).toBeInTheDocument();
  });

  it('renders the large submit button on non-medium devices', () => {
    render(<SearchForm {...searchProps} isMediumDevice={false} />);
    const submitButton = screen.getByRole('button', { name: 'Submit' });
    expect(submitButton).toBeInTheDocument();
    expect(screen.getByTestId('submit-button-large')).toBeInTheDocument();
  });

  it('renders the correct label', () => {
    render(<SearchForm {...searchProps} />);
    screen.debug();
    expect(screen.getByLabelText(/search/i)).toBeInTheDocument();
  });

  it('calls the onSearchInput function when typing in the field', () => {
    render(<SearchForm {...searchProps} />);
    fireEvent.change(screen.getByDisplayValue('React'), {
      target: { value: 'Redux' },
    });
    expect(searchProps.onSearchInput).toHaveBeenCalledTimes(1);
    expect(searchProps.onSearchInput).toHaveBeenCalledWith(expect.anything());
  });

  it('calls the onSubmit function when clicking on the button', () => {
    render(<SearchForm {...searchProps} />);
    const mockEvent = { target: { value: 'mock search value' } };
    fireEvent.submit(screen.getByRole('button'), { event: mockEvent });
    expect(searchProps.onSearchSubmit).toHaveBeenCalledTimes(1);
    expect(searchProps.onSearchSubmit).toHaveBeenCalledWith(expect.anything());
  });

  it('renders snapshot', () => {
    const { container } = render(<SearchForm {...searchProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
