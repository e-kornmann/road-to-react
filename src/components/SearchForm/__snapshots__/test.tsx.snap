// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchForm > renders snapshot 1`] = `
<form
  class="sc-hVvQAm njtLC"
>
  <div
    style="width: 100%; line-height: 20px;"
  >
    <label
      class="sc-jzuGiw laxkdV"
      for="search"
    >
      <span>
        Search:
      </span>
    </label>
    <input
      autocomplete="off"
      class="sc-kIDzkD cZhMWN"
      id="search"
      type="text"
      value="React"
    />
  </div>
  <button
    class="sc-eCtfAB sc-bPlwQk bkCXvO eXmeoh"
    data-testid="submit-button-small"
    type="submit"
  >
    Submit
  </button>
</form>
`;
